### PR TITLE
Pin electrs dependencies

### PR DESCRIPTION
**Pull request by nixbitcoin**
This commit pins electrs dependencies using [crate2nix](https://github.com/kolloch/crate2nix), updates electrs to v0.6.2 and includes a generate.sh script to recreate the expression for future updates.

Please double-check @romanz PGP fingerprint in generate.sh and the electrs 0.6.2 sha256 for security.

cc @jb55

Closes https://github.com/fort-nix/nix-bitcoin/issues/66

### CODE REVIEW

File: pkgs/electrs/generate.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+git clone https://github.com/romanz/electrs
+
+echo "Checking out latest release"
+cd /tmp/generate_electrs_expression/electrs
+latesttag=$(git describe --tags `git rev-list --tags --max-count=1`)
+git checkout ${latesttag}
+echo "Latest release is ${latesttag}"
+
+echo "Getting Roman Zeyde's PGP key"
+gpg --recv-keys 15C8C3574AE4F1E25F3F35C587CAE5FA46917CBB
```
Comment (jonasnick) [inline pkgs/electrs/generate.sh]: always better if you change state to ask the user if he/she really wants that. Something like `Do you want to import Roman Zeyde's PGP key (15C...) [yn]`

Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Added in 3227a0151f6559e6a2e5bd30e4736d7396c0399c

File: pkgs/electrs/generate.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p git gnupg
+set -e
+
+echo "Creating temporary directory"
+mkdir /tmp/generate_electrs_expression
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Fixed in 3227a0151f6559e6a2e5bd30e4736d7396c0399c

File: pkgs/electrs/generate.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+echo "Latest release is ${latesttag}"
+
+echo "Getting Roman Zeyde's PGP key"
+gpg --recv-keys 15C8C3574AE4F1E25F3F35C587CAE5FA46917CBB
+
+echo "Verifying latest release"
+git verify-tag ${latesttag}
+
+echo "Generating crate2nix expression"
+nix-shell https://github.com/kolloch/crate2nix/tarball/0.4.0 --run "crate2nix generate"
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Fixed in 3227a0151f6559e6a2e5bd30e4736d7396c0399c. Had to use a workaround with git clone and latest working commit.

File: pkgs/electrs/generate.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+echo "Fixing nix expression"
+sed -i '/lib? pkgs.lib/a\  llvmPackages ? pkgs.llvmPackages,' default.nix
+sed -i 's/(path+file.*)/(registry+https:\/\/github.com\/romanz\/electrs)/g' default.nix
+sed -i 's/nixos/nixpkgs/g' default.nix
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Fixed in 3227a0151f6559e6a2e5bd30e4736d7396c0399c

File: pkgs/electrs/generate.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+gpg --recv-keys 15C8C3574AE4F1E25F3F35C587CAE5FA46917CBB
+
+echo "Verifying latest release"
+git verify-tag ${latesttag}
+
+echo "Generating crate2nix expression"
+nix-shell https://github.com/kolloch/crate2nix/tarball/0.4.0 --run "crate2nix generate"
+
+echo "Fixing nix expression"
+sed -i '/lib? pkgs.lib/a\  llvmPackages ? pkgs.llvmPackages,' default.nix
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Added a comment in 3227a0151f6559e6a2e5bd30e4736d7396c0399c explaining these are @jb55's fixes I adopted.

File: pkgs/electrs/generate.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+sed -i 's/(path+file.*)/(registry+https:\/\/github.com\/romanz\/electrs)/g' default.nix
+sed -i 's/nixos/nixpkgs/g' default.nix
+sed -i '/crateName = "electrs";/i\        name = "electrs-${version}";' default.nix
+sed -i 's/src = (builtins.filterSource sourceFilter .\/.);/sha256 = "<insert correct hash here>";/g' default.nix
+sed -i 's/resolvedDefaultFeatures = \[ "bzip2" "default" "lz4" "snappy" "static" "zlib" "zstd" \]/resolvedDefaultFeatures = \[ "bzip2" "default" "lz4" "snappy" "static" "zlib" \]/g' default.nix
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Added a comment in 3227a0151f6559e6a2e5bd30e4736d7396c0399c explaining these are @jb55's fixes I adopted.

File: pkgs/electrs/generate.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+sed -i '/crateName = "librocksdb-sys";/a\\n        enableParallelBuilding = true;\n        buildInputs = [ pkgs.clang ];\n        LIBCLANG_PATH = "${llvmPackages.libclang}/lib";\n' default.nix
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply pkgs/electrs/generate.sh]: Added a comment in 3227a0151f6559e6a2e5bd30e4736d7396c0399c explaining these are @jb55's fixes I adopted.

**Comment by jb55** [comment]
Concept ACK

**Pull request merged**